pkg/types/clusterconfig/cluster_config.go

```diff
@@ -1116,6 +1257,84 @@ func (ng *NodeGroup) validateNodeGroup(awsClient *aws.Client, region string) err
 	return nil
 }
 
+func (cc *Config) GetNodeGroupByName(name string) *NodeGroup {
+	for _, ng := range cc.NodeGroups {
+		if ng.Name == name {
+			return ng
+		}
+	}
+	return nil
+}
+
+func (cc *Config) getNewNodeGroups(oldConfig Config) []*NodeGroup {
+	newNodeGroups := []*NodeGroup{}
+	for _, updatingNg := range cc.NodeGroups {
+		isNewNg := true
+		for _, previousNg := range oldConfig.NodeGroups {
+			if previousNg.Name == updatingNg.Name {
+				isNewNg = false
+				break
+			}
+		}
+		if isNewNg {
+			ngCopy := *updatingNg
+			newNodeGroups = append(newNodeGroups, &ngCopy)
+		}
+	}
+	return newNodeGroups
+}
+
+func (cc *Config) getRemovedNodeGroups(oldConfig Config) []*NodeGroup {
+	removedNodeGroups := []*NodeGroup{}
+	for _, previousNg := range oldConfig.NodeGroups {
+		isRemovedNg := true
+		for _, updatingNg := range cc.NodeGroups {
+			if previousNg.Name == updatingNg.Name {
+				isRemovedNg = false
+				break
+			}
+		}
+		if isRemovedNg {
+			ngCopy := *previousNg
+			removedNodeGroups = append(removedNodeGroups, &ngCopy)
+		}
+	}
+	return removedNodeGroups
+}
+
+func (cc *Config) getCommonNodeGroups(oldConfig Config) ([]*NodeGroup, []*NodeGroup) {
+	commonNewNodeGroups := []*NodeGroup{}
+	commonOldNodeGroups := []*NodeGroup{}
+	for _, previousNg := range oldConfig.NodeGroups {
+		for _, updatingNg := range cc.NodeGroups {
+			if previousNg.Name == updatingNg.Name {
+				ngNewCopy := *updatingNg
+				ngOldCopy := *previousNg
+				commonNewNodeGroups = append(commonNewNodeGroups, &ngNewCopy)
+				commonOldNodeGroups = append(commonOldNodeGroups, &ngOldCopy)
+				break
+			}
+		}
+	}
+	return commonNewNodeGroups, commonOldNodeGroups
+}
+
+func GetNodeGroupNames(nodeGroups []*NodeGroup) []string {
+	ngNames := []string{}
```

Comment:
nit: `ngNames := make([]string, 0, len(nodeGroups))` (or the index and remove the 0) (also in the function below)